### PR TITLE
Allow editing of Repositories using the form-yaml experience

### PR DESCRIPTION
**Pull request by debsmita1**
**JIRA story:**
https://issues.redhat.com/browse/ODC-6726

**Solution description:**
using the existing form for adding helm chart repositories to edit them as well

**GIF:**
https://user-images.githubusercontent.com/22490998/183068664-ff457ff2-f42a-4561-9b26-95fdce01e52f.mov

**Test coverage:**
Screenshot 2022-07-22 at 5 45 06 PM

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@debsmita1 please rebase the PR. I have reviewed the code and it looks good to me but need to test this once before adding lgtm.

**Comment by divyanshiGupta** [comment]
@debsmita1 the `Edit helm chart repository` action on the repository detail page opens the yaml tab instead of the edit form. Is this expected?

**Comment by debsmita1** [comment]
> @debsmita1 the `Edit helm chart repository` action on the repository detail page opens the yaml tab instead of the edit form. Is this expected?

Thanks for catching this. Fixed it ! PTAL

File: frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx:58]: For other resources we show the samples only when creating a new resource, not in the edit flow.

Can we archive this by checking `existingRepo` here? Btw: Why `existingRepo` instead of `resource`, isn't it the same?

```suggestion
      showSamples={!existingRepo}
```

File: frontend/packages/helm-plugin/console-extensions.json
Comment (jerolimov) [inline frontend/packages/helm-plugin/console-extensions.json:163]: Nit:

```suggestion
      "href": "/ns/:namespace/helmchartrepositories/~new?actionOrigin=addPage",
```

File: frontend/packages/helm-plugin/src/catalog/components/HelmCatalogTypeDescription.tsx
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/catalog/components/HelmCatalogTypeDescription.tsx:22]: Nit:

```suggestion
        <Link to={`/ns/${namespace}/helmchartrepositories/~new?actionOrigin=catalog`}>
```

File: frontend/packages/helm-plugin/src/actions/providers.ts
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/actions/providers.ts:88]: Do you want check here if the index exist?

Array.findIndex returns -1 if it doesn't exist and would run your if statement.

And it returns 0 if the first action is edit-resource and this would jump over your if statement.

You should also be very careful with Array.splice as it doesn't return a new array. That's only okay here because of the filter one line above. I always recommend to use Array.slice instead also it's not required here.

```suggestion
    let commonActions = getCommonResourceActions(kindObj, resource);
    const index = commonActions.findIndex((action: Action) => action.id === 'edit-resource');
    if (index => 0) {
      commonActions = commonActions.filter((action: Action) => action.id !== 'edit-resource');
      commonActions = commonActions.slice(index, 0, editChartRepository(kindObj, resource, t));
    }
    return commonActions;
```

File: frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx:89]: I understand the benefit for a developer of concatenating strings like `Create`/`Edit` plus `ProjectHelmChartRepository`/`Helm Chart Repository` - or `Add`/`Update` plus a sentence.

The problem here is that this expects that all translations have the form `$verb $noun` or `$verb sentence`, but that's not required in all languages and I would expect that makes the life of the translation team more difficult.

Using variables here also will not allow the translation team to translate "Update" and "Add".

So I recommend a much more simple (you can also call it stupid) code like this:

```suggestion
  const formDescription =
    !showScopeType && formData.scope === ProjectHelmChartRepositoryModel.kind
      ? (existingRepo
          ? t('helm-plugin~Update helm chart repository in the namespace.')
          : t('helm-plugin~Add helm chart repository in the namespace.')
        )
      : (existingRepo
          ? t('helm-plugin~Update helm chart repository.')
          : t('helm-plugin~Add helm chart repository.')
        );
```

File: frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/components/forms/HelmChartRepository/CreateHelmChartRepositoryForm.tsx:80]: Same translation issue here. Edit and Create should not be variables. This should be part of the translation. Keep it simple:

```suggestion
      ? (
        existingRepo
          ? t('helm-plugin~Edit ProjectHelmChartRepository')
          : t('helm-plugin~Create ProjectHelmChartRepository')
      )
      : (
        existingRepo
          ? t('helm-plugin~Edit Helm Chart Repository')
          : t('helm-plugin~Create Helm Chart Repository')
      );
```

File: frontend/packages/helm-plugin/src/components/list-page/HelmChartRepositoryRow.tsx
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/components/list-page/HelmChartRepositoryRow.tsx:26]: The other places uses a translation here:

```suggestion
      <TableData>{obj.spec?.disabled ? t('helm-plugin~True') : t('helm-plugin~False')}</TableData>
```

**Review by jerolimov — CHANGES_REQUESTED** [review]
I reviewed the 2nd commit and found some nits and some translation issues. Please see the comments above.

The direction of this change is fine, so I would like to approve this right now. Please check with the final reviewer that all comments are addressed or let me know if I should take a new look at this.

/approve

**Comment by jerolimov** [comment]
Found also another UI issue: The LazyActionMenu is not shown correctly above the Add/Remove from navigation button.

Because this is a general UI issue and an edge case I opened a [BZ ticket](https://bugzilla.redhat.com/show_bug.cgi?id=2115265) for this.

https://user-images.githubusercontent.com/139310/182818276-e618bbf8-9541-40ee-97bb-d3cbe881647a.mp4

https://user-images.githubusercontent.com/139310/182818264-62b069a9-2e84-46df-a923-b160cdf8d6c6.mp4

**Comment by invincibleJai** [comment]
/hold

adding hold as this PR is dependent on https://github.com/openshift/console/pull/11844, feel free to unhold it once the dependent PR is in.

cc @debsmita1

**Comment by jerolimov** [comment]
#11844 is merged

/unhold

**Review by jerolimov — CHANGES_REQUESTED** [review]
As posted on slack, the helm chart empty state page contains a link to the catalog but links to the namespace "undefined".

**Comment by vikram-raj** [comment]
For both the Edit HCR and PHCR. Form heading is the same `Edit Helm Chart Repository`. Is it expected?

**Comment by vikram-raj** [comment]
In the case of Edit HCR, the Form page URL have `undefined` as a namespace name. IMO as HCR is not a namespace scoped so can remove the `ns/:namespace` from the URL and add `/k8s/cluster/` instead. wdyt?

image

**Comment by debsmita1** [comment]
> In the case of Edit HCR, the Form page URL have `undefined` as a namespace name. IMO as HCR is not a namespace scoped so can remove the `ns/:namespace` from the URL and add `/k8s/cluster/` instead. wdyt?
> 
> image

Fixed it ! and updated the video in the PR description. PTAL

**Review by vikram-raj — APPROVED** [review]
Verified the changes and it works as expected.

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature
Comment (sanketpathak) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature:1]:
```suggestion
@helm @odc-6685
```

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature
Comment (sanketpathak) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature:61]:
```suggestion
               And user clicks on Save button
              Then user can see "ProjectHelmChartRepository" "helm-test1" details page
```

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature
Comment (sanketpathak) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-page-tabs.feature:64]: We can not have two Then statements

File: frontend/packages/helm-plugin/integration-tests/support/step-definitions/helm/helm-navigation.ts
Comment (sanketpathak) [inline frontend/packages/helm-plugin/integration-tests/support/step-definitions/helm/helm-navigation.ts:28]: Can we avoid this using timeout

Comment (debsmita1) [reply frontend/packages/helm-plugin/integration-tests/support/step-definitions/helm/helm-navigation.ts:28]: at times it takes some time for the resource to get deleted

**Review by jerolimov — APPROVED** [review]
Tested it locally and it works fine.

/approve
/lgtm

**Comment by sanketpathak** [comment]
Got this issue while testing on a cluster-bot
[Screencast from 08-06-2022 06:11:55 PM.webm](https://user-images.githubusercontent.com/10252230/183393162-883b3f37-4a9d-4869-967f-dfe93ba2d701.webm)

**Comment by sanketpathak** [comment]
/label qe-approved
[Screencast from 08-09-2022 05:36:02 PM.webm](https://user-images.githubusercontent.com/10252230/183643710-4fc079ab-a54f-4371-a7c5-2022e3cd457b.webm)

**Pull request merged**
